src/mapwindow.jl

```diff
@@ -61,8 +61,8 @@ and then `mapwindow(f, img, (m,n))` should filter at the 75th quantile.
 
 See also: [`imfilter`](@ref).
 """
-function mapwindow(f, img, window; border="replicate",
-                   indices=default_imginds(img, window, border), callmode=:copy!)
+function mapwindow(f::F, img, window; border="replicate",
+                   indices=default_imginds(img, window, border), callmode=:copy!) where F
```

Comment:
So `f::F where F` is more friendly to precompilation than `f::Any`?

Reply:
As you may know, it forces specialization on `f`; while Julia typically specializes for each argument type, potentially unlimited "abstract" types like `Function`, `Type`, and `Tuple` have heuristics that cause Julia to sometimes choose to compile a generic version rather than a specific one. Historically the main reason to force specialization was better runtime performance, at the cost of *increased* latency (because you have to do more compilation for each specialization). However, loss of specialization means loss of backedges and thus precompilability, so its a bit of a tradeoff.

It might be worth experimenting here a bit more before merging this, just to see if we can get the same benefit by other means. I didn't poke at this point extensively, I just noticed an inference failure due to lack of specialization. But the main reason I changed it is that `mapwindow` is pretty performance-sensitive, and it makes sense that at least `mapwindow_kernel` *should* be specialized on `f`. This PR doesn't technically guarantee that, though it seems the current heuristics must be making that happen anyway given the changes here. Like I said, probably worth poking a bit more before we merge this. I've got a few other priorities first (I did this as part of testing out major changes to SnoopCompile, and I want to get that done before coming back to this).